pandas/plotting/_core.py

```diff
@@ -1597,6 +1598,11 @@ def __init__(self, data, return_type='axes', **kwargs):
                 "return_type must be {None, 'axes', 'dict', 'both'}")
 
         self.return_type = return_type
+        if isinstance(data, ABCDataFrame):
+            data = data.select_dtypes(include='number')
+        elif is_datetimelike(data):
```

Comment:
shouldn't this be for any non-DataFrame?

Reply:
Hmm, I don't follow - I must be missing something. My thinking is that 

```python
if isinstance(data, ABCDataFrame):
    data = data.select_dtypes(include='number')
```

will skip plotting any non-numerical, datetime-like columns, and 

```python
elif is_datetimelike(data):
    raise ValueError(...)
```

will catch any non-DataFrames (e.g. `Series`, `DatetimeIndex`, etc.) with datetime-like data where making a `boxplot` doesn't make sense. 

Is this not the case?

Reply:
Is this condition specific to box plots? I understand your thinking but it's somewhat orthogonal to the original issue. Might be better served as a separate issue with a more comprehensive change unless it is specific to this type of plot